packages/apollo-cli/src/commands/queries/extract.ts

```diff
@@ -0,0 +1,80 @@
+import 'apollo-codegen-core/lib/polyfills';
+import { Command, flags } from '@oclif/command';
+import * as Listr from 'listr';
+import { print } from 'graphql';
+import { sha512 } from 'js-sha512';
+import * as fs from 'fs';
+
+import { loadQueryDocuments } from 'apollo-codegen-core/lib/loading';
+
+import { engineFlags } from '../../engine-cli';
+import { resolveDocumentSets } from '../../config';
+import { loadConfigStep } from '../../load-config';
+
+export default class ExtractQueries extends Command {
+  static description = 'Extracts queries';
+
+  static flags = {
+    help: flags.help({
+      char: 'h',
+      description: 'Show command help',
+    }),
+    config: flags.string({
+      description: 'Path to your Apollo config file',
+    }),
+    queries: flags.string({
+      description:
+        'Path to your GraphQL queries, can include search tokens like **',
+    }),
+    json: flags.boolean({
+      description: 'Output result as JSON',
+    }),
+    ...engineFlags,
+
+    tagName: flags.string({
+      description:
+        'Name of the template literal tag used to identify template literals containing GraphQL queries in Javascript/Typescript code',
+      default: 'gql',
+    }),
+  };
+
+  async run() {
+    const { flags } = this.parse(ExtractQueries);
+
+    const tasks: Listr = new Listr([
+      loadConfigStep(flags, false),
+      {
+        title: 'Resolving GraphQL document sets',
+        task: async (ctx, task) => {
+          ctx.documentSets = await resolveDocumentSets(ctx.config, false);
+          const operations = loadQueryDocuments(
+            ctx.documentSets[0].documentPaths,
+            flags.tagName,
+          );
+          task.title = `Scanning for GraphQL queries (${
+            operations.length
+          } found)`;
+          // XXX send along file information as well
+          ctx.operations = operations.map(doc => ({ document: print(doc) }));
+        },
+      },
+      {
+        title: 'Generating hashes',
+        task: async ctx => {
+          ctx.mapping = {};
+          (ctx.operations as Array<{document: string}>).forEach(({ document }) => {
+            ctx.mapping[sha512.update(document).hex()] = document;
+          });
+        },
+      },
+      {
+        title: 'Outputing manifest',
+        task: async ctx => {
+          fs.writeFileSync('manifest.json', JSON.stringify(ctx.mapping));
```

Comment:
This should really be an option before this gets merged. People will riot otherwise 😂

Reply:
😆 Totally! What do you think about the default being output to stdout and then having an option to specify the file with `--output`? Or should it automatically send it to the manifest?

Reply:
That would be one option, but consider the rest of the cli that just takes the output path as the last arg without a flag.